src/librados/IoCtxImpl.cc

```diff
@@ -1103,6 +1103,7 @@ int librados::IoCtxImpl::watch(const object_t& oid,
   set_sync_op_version(objver);
 
   if (r < 0) {
+    delete linger_op->watch_context;
```

Comment:
should this be part of linger_cancel() so there's no race between deleting it and notifies trying te use it?